lib/hatchet/api_rate_limit.rb

```diff
@@ -0,0 +1,40 @@
+# Wraps platform-api and adds API rate limits
+#
+# Instead of:
+#
+#     platform_api.pipeline.create(name: @name)
+#
+# Use:
+#
+#     api_rate_limit = ApiRateLimit.new(platform_api)
+#     api_rate_limit.call.pipeline.create(name: @name)
+#
+class ApiRateLimit
+  def initialize(platform_api)
+    @platform_api = platform_api
+    @capacity = 1
+    @called   = 0
+  end
+
+
+  # Sleeps for progressively longer when api rate limit capacity
+  # is lower.
+  #
+  # Unfortunatley `@platform_api.rate_limit` is an extra API
+  # call, so by checking our limit, we also are using our limit 😬
+  # to partially mitigate this, only check capacity every 5
+  # api calls, or if the
```

Comment:
> or if the... 

??

Reply:
we may never know 🤔 I'll fix that. Was going to say if there are fewer than 1000 requests in the bucket.